go/store/datas/database_common.go

```diff
@@ -616,14 +751,27 @@ func (db *database) doUpdateWorkingSet(ctx context.Context, datasetID string, wo
 		}()
 	}
 
-	return tryCommitErr
+	commitDS, err = db.GetDataset(ctx, commitDS.ID())
+	if err != nil {
+		return Dataset{}, Dataset{}, err
+	}
+
+	workingSetDS, err = db.GetDataset(ctx, workingSetDS.ID())
+	if err != nil {
+		return Dataset{}, Dataset{}, err
+	}
```

Comment:
Is this potentially subtly racy in the same way that the commit logic used to be? Could it theoretically get a smeared view of HEAD vs. workingset and should it work with the same root map across both calls instead?

Reply:
Good call, fixed.